sdk/examples/intkey_go/src/sawtooth_intkey/main.go

```diff
@@ -51,6 +52,7 @@ func main() {
 	handler := intkey.NewIntkeyHandler(prefix)
 	processor := processor.NewTransactionProcessor(endpoint)
 	processor.AddHandler(handler)
+	processor.ShutdownOnSignal(syscall.SIGINT, syscall.SIGTERM)
```

Comment:
Technically there isn't. I wasn't sure whether it should or not and am open to hearing other viewpoints.

My thought process was: Since this is an SDK it should be general and limit the assumptions made about how it is used. That includes leaving it up to the main package/function to decided how and when to shutdown. By making the Shutdown() function part of the API and making it easy to setup signal handling with ShutdownOnSignal(), the API allows the following use case while still making it easy to handle SIGINT in the simple case:
1. Starting more than one transaction processor in a single process. This would require a custom signal handler which would call Shutdown() on all the processors running.
2. Not shutting down on Ctrl+C or perhaps shutting down on reading from a channel or some other event.
3. Shutting down on signals other than or in addition to SIGINT